pkg/coscheduling/coscheduling_test.go

```diff
@@ -18,50 +18,83 @@ package coscheduling
 
 import (
 	"context"
-	"fmt"
 	"testing"
 	"time"
 
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/clock"
-	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
 	clientsetfake "k8s.io/client-go/kubernetes/fake"
-	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultbinder"
-	fwkruntime "k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+	"k8s.io/client-go/tools/events"
 	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
 	st "k8s.io/kubernetes/pkg/scheduler/testing"
-	"k8s.io/kubernetes/pkg/scheduler/util"
 
-	"sigs.k8s.io/scheduler-plugins/pkg/apis/config"
-	// Ensure scheme package is initialized.
 	_ "sigs.k8s.io/scheduler-plugins/pkg/apis/config/scheme"
+	"sigs.k8s.io/scheduler-plugins/pkg/coscheduling/core"
+	fakepgclientset "sigs.k8s.io/scheduler-plugins/pkg/generated/clientset/versioned/fake"
+	pgformers "sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions"
+	pgutil "sigs.k8s.io/scheduler-plugins/pkg/util"
+	testutil "sigs.k8s.io/scheduler-plugins/test/util"
 )
 
-// FakeNew is used for test.
-func FakeNew(clock util.Clock, args config.CoschedulingArgs, stop chan struct{}) (*Coscheduling, error) {
-	cs := &Coscheduling{
-		clock: clock,
-		args:  args,
-	}
-	go wait.Until(cs.podGroupInfoGC, time.Duration(cs.args.PodGroupGCIntervalSeconds)*time.Second, stop)
-	return cs, nil
-}
-
 func TestLess(t *testing.T) {
 	labels1 := map[string]string{
-		PodGroupName:         "pg1",
-		PodGroupMinAvailable: "3",
+		pgutil.PodGroupLabel: "pg1",
 	}
 	labels2 := map[string]string{
-		PodGroupName:         "pg2",
-		PodGroupMinAvailable: "5",
+		pgutil.PodGroupLabel: "pg2",
+	}
+	labels3 := map[string]string{
+		pgutil.PodGroupLabel: "pg3",
+	}
+	labels4 := map[string]string{
+		pgutil.PodGroupLabel: "pg4",
 	}
 
-	var lowPriority, highPriority = int32(10), int32(100)
 	t1 := time.Now()
-	t2 := t1.Add(time.Second)
+	t2 := time.Now().Add(time.Second)
+	t3 := time.Now().Add(2 * time.Second)
+	t4 := time.Now().Add(3 * time.Second)
+	t5 := time.Now().Add(-2 * time.Second)
+	t6 := time.Now().Add(-1 * time.Second)
+
+	t.Log(t1)
+	t.Log(t2)
+	t.Log(t3)
+	t.Log(t4)
+	t.Log(t5)
+	t.Log(t6)
```

Comment:
Do you mind using a for loop to simplify the logic, also applies to the logic below of instantiating PGs.

Reply:
done